qiskit/circuit/parameterexpression.py

```diff
@@ -490,6 +499,74 @@ def __eq__(self, other):
             return len(self.parameters) == 0 and complex(self._symbol_expr) == other
         return False
 
+    def __lt__(self, other: ParameterValueType) -> bool:
+        """Check if this parameter expression is less than another parameter expression
+           or a fixed value (only if this is a bound expression).
+        Args:
+            other (ParameterExpression or a number):
+                Parameter expression or numeric constant used for comparison
+        Raises:
+            TypeError:
+                - If comparison to number with unbound parameters.
+                - If comparison to type of other object unsupported.
+        Returns:
+            bool: result of the comparison
+        """
+        if isinstance(other, ParameterExpression):
+            if HAS_SYMENGINE:
+                from sympy import sympify
+
+                return sympify(self._symbol_expr).__lt__(sympify(other._symbol_expr))
+            else:
+                return self._symbol_expr.__lt__(other._symbol_expr)
```

Comment:
These don't do what you think they do; they don't always return Boolean values. Also, in general it's better to use the Python operations, and let Python handle the numerics; this way can fail if the left-hand side doesn't evaluate `__lt__`, but the right hand does evaluate `__gt__`, whereas if you'd used `<` in that situation, Python would evaluate it correctly.

For not returning a Boolean, see:
```python
In [1]: import sympy
   ...: x = sympy.Symbol("x")
   ...: x + 1 > x
Out[1]: x + 1 > x
```